pipeline/metadata/flatten.py

```diff
@@ -366,13 +370,52 @@ def _process_satellite_v2(self, scan: Any,
         'error': scan.get('error', None),
         'anomaly': scan['anomaly'],
         'success': not scan['connect_error'],
+        'controls_failed': not scan['passed_control'],
         'received': None,
         'rcode': [],
         'measurement_id': random_measurement_id
     }
     received_ips = scan.get('response')
     yield from self._process_received_ips(row, received_ips)
 
+  def _process_satellite_v2_responses(
+      self, scan: Any, random_measurement_id: str) -> Iterator[Row]:
+    """Process a line of Satellite response data.
+
+      Args:
+        scan: a loaded json object containing the parsed content of the line
+        random_measurement_id: a hex id identifying this individual measurement
+
+      Yields:
+        Rows
+    """
+    row = {
+        'domain': scan['test_url'],
+        'category': self.category_matcher.match_url(scan['test_url']),
+        'ip': scan['vp'],
+        'anomaly': None,
+        'success': not scan['connect_error'],
+        'controls_failed': not scan['passed_control'],
+        'measurement_id': random_measurement_id
+    }
+    responses = scan.get('response', [])
+    if responses:
```

Comment:
change `resp` -> `response` in this whole section, just for easier reading.

Reply:
Done